contracts/staking/P12MineUpgradeable.sol

```diff
@@ -284,8 +284,8 @@ contract P12MineUpgradeable is
   */
   function queueWithdraw(address lpToken, uint256 amount) public virtual override whenNotPaused nonReentrant {
     uint256 pid = getPid(lpToken);
-    PoolInfo memory pool = poolInfos[pid];
-    UserInfo memory user = userInfo[pid][msg.sender];
+    PoolInfo storage pool = poolInfos[pid];
```

Comment:
`pool` is not modified here, so is using memory okay?

Reply:
If the storage is not used, the state of the pool modified in the _checkPoint function cannot take effect, because the memory modification is equivalent to a value reference

Reply:
> _checkPoint

But in `_checkPoint`, variable `pool` will be loaded from storage again. variable `pool` in this context is not modified.

Reply:
```
contract Storage {
    Info[] public infos;
    
    struct Info{
        string name;
    }
    
    
    function setInfo()public {
        Info memory info = Info("River");
        infos.push(info);
    }


   
   function getInfo1(string memory _name)public returns(string memory){
        Info memory info = infos[0];
        modifyInfo(_name);
        return info.name;
   }

    
    function getInfo2(string memory _name)public returns(string memory){
        modifyInfo(_name);
        Info memory info = infos[0];
        return info.name;
   }

   function getInfo3(string memory _name)public returns(string memory){
        Info storage info = infos[0];
        modifyInfo(_name);
        return info.name;
   }
   
    function modifyInfo(string memory _name)internal {
       Info storage info = infos[0];
       info.name = _name;
   }
}
```

This simple demo I think can explain the problem very well

Reply:
![image](https://user-images.githubusercontent.com/34837173/184830085-a8afe577-f702-4855-bef7-9911812f8549.png)